### DISCUSSION TITLE
You might want to check it out

### DISCUSSION THREAD

**Discussion by benjamin-henry** [GiorgosXou/NeuralNetworks#7] [Ideas]
Hi. I saw Softmax was not implemented. I wrote an mlp library for Arduino, you can check it out, I think you will find useful things in it.

https://github.com/benjamin-henry/MLP_C

Cheers.

**Comment by GiorgosXou** [comment]
> Hi.

Hello. 👋
> I wrote an mlp library for Arduino, you can check it out

That's dope bro, i'm definitely gonna check it out, **Thanks!**
>  I think you will find useful things in it.

From a brief look that I had into your code, **i've already found many intresting things [...]**
> I saw Softmax was not implemented. 

**True**, when i first saw softmax i was in a hurry to publish my next new version and softmax was kind of confusing to me at that time *(actually even now a bit)*. Until today that you brought it to my attention, I had a fresh look to it once again and thankfully i found out about those videos who really helped me out understanding it too: 

* [Softmax Function in Deep Learning](https://www.youtube.com/watch?v=EuZZ6plg2Tk)
* [The Softmax : Data Science Basics](https://www.youtube.com/watch?v=8ps_JEW42xs)
* [Derivative of Sigmoid and Softmax Explained Visually](https://www.youtube.com/watch?v=gRr2Q97XS2g)

**But still i don't get  the derivative part of it that well [...]**
> Cheers.

 **BTW**, we need people who know a bit suff about this topic, which have an interest and they are willing to share their knowledge with others *(cause this subject is kind of dificult [at least from my perspective, etc.])*. So.. if you want, **i would be really happy to see you in our [discord server](https://discord.com/invite/R2RVeD3gY8)**  *(which is kinda new but anyways lol)*, even just for fun or maybe for sharing any intresting links such this one or whatever, whenever you like. 😃 
 
**Cheers mate!** 😄

**Comment by GiorgosXou** [comment]
Just a while ago, I had a deeper look into your code. **Again, thank you!**  it gave me the things I always search for: 
* A programmatic perspective *(and not just pure maths)*. 
* The confidence to implement things i wasn't that sure i could implement well enough yet.
* and the amount of motivation & support i needed (: 

<sub><sup>[Discord server](https://discord.com/invite/R2RVeD3gY8) ?</sup></sub>

**Reply by benjamin-henry** [reply]
Hi.
I wrote a function to save a mlp on esp32 eeprom. I am also implementing rmsprop optimizer.
This will be pushed in few days.
May the matrix be on your side 😎
